From 1ec1688fcb2ecc8a4bf865b6f61d65bf8410a157 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 6 Mar 2023 16:34:39 +0000
Subject: [PATCH] overlays: Remove lirc-rpi from media-center

lirc-rpi was retired years ago, suggesting that this overlay is not
being used. Before deleting outright, remove the lirc-rpi element.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README             | 12 -----
 .../dts/overlays/media-center-overlay.dts     | 50 +------------------
 2 files changed, 1 insertion(+), 61 deletions(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2774,18 +2774,6 @@ Params: speed                   Display
         xohms                   Touchpanel sensitivity (X-plate resistance)
         swapxy                  Swap x and y axis
         backlight               Change backlight GPIO pin {e.g. 12, 18}
-        gpio_out_pin            GPIO for output (default "17")
-        gpio_in_pin             GPIO for input (default "18")
-        gpio_in_pull            Pull up/down/off on the input pin
-                                (default "down")
-        sense                   Override the IR receive auto-detection logic:
-                                 "0" = force active-high
-                                 "1" = force active-low
-                                 "-1" = use auto-detection
-                                (default "-1")
-        softcarrier             Turn the software carrier "on" or "off"
-                                (default "on")
-        invert                  "on" = invert the output pin (default "off")
         debug                   "on" = enable additional debug messages
                                 (default "off")
 
--- a/arch/arm/boot/dts/overlays/media-center-overlay.dts
+++ b/arch/arm/boot/dts/overlays/media-center-overlay.dts
@@ -73,62 +73,14 @@
 		};
 	};
 
-	fragment@4 {
-		target-path = "/";
-		__overlay__ {
-			lirc_rpi: lirc_rpi {
-				compatible = "rpi,lirc-rpi";
-				pinctrl-names = "default";
-				pinctrl-0 = <&lirc_pins>;
-				status = "okay";
-
-				// Override autodetection of IR receiver circuit
-				// (0 = active high, 1 = active low, -1 = no override )
-				rpi,sense = <0xffffffff>;
-
-				// Software carrier
-				// (0 = off, 1 = on)
-				rpi,softcarrier = <1>;
-
-				// Invert output
-				// (0 = off, 1 = on)
-				rpi,invert = <0>;
-
-				// Enable debugging messages
-				// (0 = off, 1 = on)
-				rpi,debug = <0>;
-			};
-		};
-	};
-
-	fragment@5 {
-		target = <&gpio>;
-		__overlay__ {
-			lirc_pins: lirc_pins {
-				brcm,pins = <6 5>;
-				brcm,function = <1 0>; // out in
-				brcm,pull = <0 1>; // off down
-			};
-		};
-	};
-
 	__overrides__ {
 		speed =     <&rpidisplay>,"spi-max-frequency:0";
 		rotate =    <&rpidisplay>,"rotate:0";
 		fps =       <&rpidisplay>,"fps:0";
-		debug =     <&rpidisplay>,"debug:0", 
-		            <&lirc_rpi>,"rpi,debug:0";
+		debug =     <&rpidisplay>,"debug:0";
 		xohms =     <&rpidisplay_ts>,"ti,x-plate-ohms;0";
 		swapxy =    <&rpidisplay_ts>,"ti,swap-xy?";
 		backlight = <&rpidisplay>,"led-gpios:4",
 		            <&rpi_display_pins>,"brcm,pins:0";
-
-		gpio_out_pin =  <&lirc_pins>,"brcm,pins:0";
-		gpio_in_pin =   <&lirc_pins>,"brcm,pins:4";
-		gpio_in_pull =  <&lirc_pins>,"brcm,pull:4";
-
-		sense =         <&lirc_rpi>,"rpi,sense:0";
-		softcarrier =   <&lirc_rpi>,"rpi,softcarrier:0";
-		invert =        <&lirc_rpi>,"rpi,invert:0";
 	};
 };
